backend/src/main/java/co/inventorsoft/scripty/config/AuthorizationServerConfig.java

```diff
@@ -0,0 +1,86 @@
+package co.inventorsoft.scripty.config;
+
+import java.util.Arrays;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerSecurityConfigurer;
+import org.springframework.security.oauth2.provider.token.TokenEnhancerChain;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.oauth2.provider.token.store.JwtAccessTokenConverter;
+
+/**
+ * @author lzabidovsky 
+ */
+@Configuration
+@EnableAuthorizationServer
+public class AuthorizationServerConfig extends AuthorizationServerConfigurerAdapter {
+
+	@Value("${security.jwt.client-id}")
+	private String clientId;
+
+	@Value("${security.jwt.client-secret}")
+	private String clientSecret;
+
+	@Value("${security.jwt.resource-ids}")
+	private String resourceIds;
+
+	@Value("${token.access.validity.seconds}")
+	private int accessValidity;
+
+	@Value("${token.refresh.validity.seconds}")
+	private int refreshValidity;
+
+	@Autowired
+	private TokenStore tokenStore;
+
+	@Autowired
+	private JwtAccessTokenConverter accessTokenConverter;
+
+	@Autowired
+	private AuthenticationManager authenticationManager;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	@Autowired
+	private UserDetailsService userDetailsService;
+
+	@Override
+	public void configure(ClientDetailsServiceConfigurer configurer) throws Exception {
+		configurer
+			.inMemory()
+			.withClient(clientId)
+			.secret(passwordEncoder.encode(clientSecret))
```

Comment:
Why should we encode secret here? Were it be encoded on frontend to send request? I don't see an additional security by doing this. Here are the reasons:
1) we still have secret as plain text in config file;
2) if we want to send plain secret from ui and then encode it on backend to verify with our config secret, we still need to store client secret on ui as plain text;
3) if we will use encoded version on both ends (backend and ui) then it makes no difference from using plain version, cause we will do simple equality check.

Reply:
Spring Security 5 (Spring Boot 2) uses a modernized password storage. Prior to Spring Security 5.0 the default PasswordEncoder was NoOpPasswordEncoder which required plain text passwords. The default PasswordEncoder is now something like BCryptPasswordEncoder. NoOpPasswordEncoder is deprecated. The easiest modern way to configure ClientDetailsServiceConfigurer is to use passwordEncoder.encode

Reply:
I understand that and it's fine for user passwords. However, I don't see the how  it improves security in our case. Let's try to see what is the flow. We have clientId and clientSecret we need to pass to obtain access token. This should be stored on the frontend config (that means it's accessible by all, cause js can be easily seen in browser by everyone). If we encode clientSecret with Bcrypt here that means we have two options:
1) pass clientSecret in plain text and then hash it with bcrypt and compare with already encoded client secret on backend;
2) pass encoded clientSecret from the frontend and simply check with backend one for equality.
In both cases clientSecret is exposed to everyone and there is no additional security in it. So I don't see a reason do it. Does auth0 works like (1) option I described?